### ISSUE TITLE
[Broken Website] Github has grey colors

### ISSUE THREAD

**Issue by kkoomen** [darkreader/darkreader#4182]
**Website**
https://github.com

**How to Reproduce**

Visit any repository.

**Expected behavior**

Before, GitHub was dark as it should, but I guess they changed some styles, so .. now things don't work as expected anymore.

**Actual behavior**
Everything is grey instead of black.

**Screenshots**

<img width="1156" alt="image" src="https://user-images.githubusercontent.com/10693490/99522347-6f568200-298d-11eb-9e2e-ad046220eed9.png">

<img width="1516" alt="image" src="https://user-images.githubusercontent.com/10693490/99522520-ad53a600-298d-11eb-9d2c-a1441996ca1c.png">

**System info:**

- OS: MacOS
- Browser: Firefox 82.0.3 (64-bit)
- Darkreader Version: 4.9.23

**Additional context**
N/A

**Comment by Gusted** [comment]
That's indeed very broken. Will have a look when have time, I use this personally so high priority.

**Comment by ric-evans** [comment]
Good to see this being worked on. This is also happening in light mode

**Comment by jenabaivab** [comment]
Man i was furiously changing browsers thinking something was wonky in Edge. Glad I'm not alone.

**Comment by Gusted** [comment]
Some further debugging suggest the same underlying bug is as #3241 and that Github is working on a native dark mode :eyes: A serious rewrite of the current variables implementation should fix this bug. @alexanderby any proposed idea, else I will open up a PR with some proposed idea's.

Regards,
Gusted

**Comment by paniash** [comment]
Thank God, I'm not alone. I thought that the firefox update to version 83 caused this!

**Comment by Gusted** [comment]
[4.9.24](https://github.com/darkreader/darkreader/commit/ea3e94b73913879e3e4b31a8164f3472bdf19803) is under review which contains #4194

**Comment by alfoltran** [comment]
Sad. 😢 For now I am using the following workaround:

```css
.gh-header, .timeline-comment, .discussion-timeline-actions, .thread-subscription-status {
    background-color: #16191c !important;
}
.project-header, .bg-white{
    background-color: #16191c !important;
}
```

Adjust the color to your settings. Insert via Developer Tools:
![image](https://user-images.githubusercontent.com/5090235/99715083-2a157b80-2a85-11eb-8cc9-021c1a05cfd3.png)

Put it in the GitHub (.com) section:
![image](https://user-images.githubusercontent.com/5090235/99715205-4b766780-2a85-11eb-89fc-94be72490b8d.png)

I in particular was very upset with this problem. 😬

**Comment by alexanderby** [comment]
The fix is already available for Firefox. Unfortunately Chrome asked to submit some data usage policy and it seems like it will take some time for them to review it.

**Comment by alfoltran** [comment]
> 
> 
> The fix is already available for Firefox. Unfortunately Chrome asked to submit some data usage policy and it seems like it will take some time for them to review it.

Are you sure? I have version 4.9.24 and it is not working.

**Comment by Gusted** [comment]
> Are you sure? I have version 4.9.24 and it is not working.

Click on reset on the devtools.

**Comment by alfoltran** [comment]
> Click on reset on the devtools.

Great! It worked. 😁

**Comment by LukasmanMHdude** [comment]
I have version 4.9.22 and I'm still getting this error. I've updated my extensions through the "manage extensions" page on g chrome. Fix seems so sometimes work and sometimes not.

**Comment by Gusted** [comment]
> I have version 4.9.22

Wait for 4.9.24 that contains the fix. Chrome is still reviewing the extension.

**Comment by privacyguy123** [comment]
Reset on DevTools doesnt' work - I'm on 4.9.24 with alleged "fix" ... Twitter looks the same way.

**Comment by kkoomen** [comment]
@Gusted This is still not fixed. I'm using the same browser as mentioned before, but now I'm using DarkReader v4.9.24.

Now the problem is  that everything is *too* dark:

<img width="1513" alt="image" src="https://user-images.githubusercontent.com/10693490/99824376-8c39b380-2b4d-11eb-8a98-44c05e7d51c5.png">

**Comment by privacyguy123** [comment]
Here's what I see, default settings.

![image](https://user-images.githubusercontent.com/50337995/99831503-14708680-2b57-11eb-9538-c04447a86fe3.png)

**Comment by Gusted** [comment]
I still have the correct site.
If you open devtools and search for 'github.*.com'

Does it have under the CSS section this kind of CSS 
![image](https://user-images.githubusercontent.com/25481501/99840045-15f87980-2b6c-11eb-9591-a21decdc351b.png)
that´s the fix and should be their.

Also it might get fixed if you enable 'Synchronize Sites fixes'.



    Click Dark Reader icon.
    Click Dev tools (in the bottom-right corner).
    Click Preview new design.


 ⚙Settings -> Manage settings -> Synchronize Sites fixes. and Restart the browser or even the OS(as sometime browsers still has an cached state of the extension)

Regards,
Gusted

**Comment by privacyguy123** [comment]
I have always had all those options on and restart my browser and computer many times in a day - these fixes do not show on my Dev Tools.

**Comment by Gusted** [comment]
Then I suggest as an workaround,
copy paste the content of [the latest dynamic-theme-fixes.config](https://raw.githubusercontent.com/darkreader/darkreader/master/src/config/dynamic-theme-fixes.config) and pase them into the devtools and press on 'Apply', while I have an look later for an possible reason why the devtools didn't update.

Regards,
Gusted

**Comment by kkoomen** [comment]
@Gusted I copy-pasted your config and that didn't change anything at all. Sounds like you're not using the same way as we do.

**Comment by linsui** [comment]
With the fix

![sshot-027](https://user-images.githubusercontent.com/36977733/99866265-b28c3d00-2ba7-11eb-9a2d-a1453671613d.png)

With the css removed

![sshot-026](https://user-images.githubusercontent.com/36977733/99866270-b750f100-2ba7-11eb-89e5-629c54d6c120.png)

**Comment by LukasmanMHdude** [comment]
> With the fix
> 
> ![sshot-027](https://user-images.githubusercontent.com/36977733/99866265-b28c3d00-2ba7-11eb-9a2d-a1453671613d.png)
> 
> With the css removed
> 
> ![sshot-026](https://user-images.githubusercontent.com/36977733/99866270-b750f100-2ba7-11eb-89e5-629c54d6c120.png)

getting this issue as well

**Comment by LukasmanMHdude** [comment]
![image](https://user-images.githubusercontent.com/69743171/99868064-0da43c80-2b74-11eb-95b2-d11c558b759c.png)

**Comment by Gusted** [comment]
@kkoomen @LukasmanMHdude  @linsui  @privacyguy123 

Do you have custom settings for github.com e.g. higher/lower brightness/contrast? 
As the screenshot above me looks like the fix doesn't care about the custom settings and apply the default color with default settings.

Regards,
Gusted

**Comment by kkoomen** [comment]
> Do you have custom settings for github.com e.g. higher/lower brightness/contrast?

Not specifically for GitHub. I have some adjustments for the contrast globally (-10), that's it.

**Comment by Gusted** [comment]
Welp, guess alexander was right https://github.com/darkreader/darkreader/issues/4204#issuecomment-731393609

I will fiddle around the code and see if the 'variables' can be affected the settings.

**Comment by kkoomen** [comment]
But even with default settings I can see some small differences that don't look right. Have a look at my settings now (not sure if these are default) 

<img width="324" alt="image" src="https://user-images.githubusercontent.com/10693490/99869874-57b60e00-2bc6-11eb-80c8-5b24958959ea.png">

with results in 

<img width="1677" alt="image" src="https://user-images.githubusercontent.com/10693490/99869879-5edd1c00-2bc6-11eb-837a-c8dad42be8c9.png">

GitHub is dark-blue ish with some dark grey. Definitely not ok.

**Comment by linsui** [comment]
> Not specifically for GitHub. I have some adjustments for the contrast globally (-10), that's it.

The same here. The gray background won't change with the contrastness setting while other part get darker.

**Comment by Gusted** [comment]
I've opened up an PR(#4220) to fix it.
I've also commented an zip file that can be used to sideload the extension if you like to.

Regards,
Gusted

**Comment by privacyguy123** [comment]
How does one sideload an addon into FireFox with no more sideload option?

**Comment by Andre601** [comment]
Not quite sure if this is any related to dark reader, but is this (random) behaviour where the Repository list on the start page has a scrolling bar a bug from Darkreader or could it be a basic GitHub behaviour?

Example
![image](https://user-images.githubusercontent.com/11576465/99890024-49660180-2c5b-11eb-91ca-ae742a4776af.png)

**Comment by Andre601** [comment]
I hope 4.9.24 will soon be available on the Edge store (I assume it awaits verification?)

I, for now, applied the fix manually. So thanks for this! Appreciate it a lot.

**Comment by LukasmanMHdude** [comment]
> @kkoomen @LukasmanMHdude @linsui @privacyguy123
> 
> Do you have custom settings for github.com e.g. higher/lower brightness/contrast?
> As the screenshot above me looks like the fix doesn't care about the custom settings and apply the default color with default settings.
> 
> Regards,
> Gusted

I changed my settings to show the contrast, but the issue persists with normal settings, its just less noticeable
![image](https://user-images.githubusercontent.com/69743171/99890705-f66a5b00-2c16-11eb-8675-99d573e377da.png)

**Comment by privacyguy123** [comment]
Worth noting these color problems happen on other websites, but I understand this ticket is about Github specifically which is still broken for me after all these "fixes."

**Comment by Gusted** [comment]
> Worth noting these color problems happen on other websites.

Could you point out those site?

> but I understand this ticket is about Github specifically which is still broken for me after all these "fixes."

https://github.com/darkreader/darkreader/issues/4204#issuecomment-731744661

**Comment by privacyguy123** [comment]
Twitter which I and many others would use on the daily I imagine! Only after a page refresh though, which is a strange step to reproduce indeed.

**Comment by Andre601** [comment]
Twitter has a native dark theme. Just saying

**Comment by Gusted** [comment]
> Twitter which I and many others would use on the daily I imagine! Only after a page refresh though, which is a strange step to reproduce indeed.

Not reproduce-able, please open up a new issue for this.

**Comment by privacyguy123** [comment]
> 
> 
> > Twitter which I and many others would use on the daily I imagine! Only after a page refresh though, which is a strange step to reproduce indeed.
> 
> Not reproduce-able, please open up a new issue for this.

Mind boggling. None of this happened previous to this DR version, btw.

**Comment by Andre601** [comment]
Since this is related (And create a new issue just for this is redundant in my opinion) do I post it here.

While 4.9.24 did fix grey colors in places like the Repository and issues, does it still happen for the [notification-tab (Beta Version)?](https://github.com/notifications):
![image](https://user-images.githubusercontent.com/11576465/100125043-a260b000-2e7c-11eb-984e-9e01a0668b21.png)

**Comment by Gusted** [comment]
Yeah, github has added more elements that uses the old variable and that weren't a few days back and didn't got fixed.

**Comment by kkoomen** [comment]
@Gusted Which version should contain the fix? I am using v4.9.24 right now on Firefox and am still facing the problem:

<img width="1323" alt="image" src="https://user-images.githubusercontent.com/10693490/100185907-ef499280-2f1f-11eb-952b-5aec49c1f952.png">

**Comment by Gusted** [comment]
Wait for v4.9.25 [https://github.com/darkreader/darkreader/commit/8377c32c515bd261eced3dbfb6c95f9d6aa79065#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519] to be released into the webstores.

**Comment by caneleex** [comment]
just to clarify, i'm running 4.9.25, is this how the comment box part is supposed to look? ![img](https://i.imgur.com/PPQmakG.png)
this is how it looks without dark mode https://i.imgur.com/DeBmvO9.png

**Comment by alexanderby** [comment]
I think yes, this is a temporary fix. Later we will try improving CSS variables support.

**Comment by kkoomen** [comment]
> just to clarify, i'm running 4.9.25, is this how the comment box part is supposed to look? ![img](https://camo.githubusercontent.com/6264f11add6dadd7d5b290441916785a3346c8d6c8976d783876e6cd23159947/68747470733a2f2f692e696d6775722e636f6d2f5050516d616b472e706e67)
> 
> this is how it looks without dark mode https://i.imgur.com/DeBmvO9.png

Can you send a screenshot with v4.9.25 with a contrast of -20? I doubt if that still looks okay.

**Comment by kkoomen** [comment]
<img width="1182" alt="image" src="https://user-images.githubusercontent.com/10693490/100352051-50d33500-2fe4-11eb-91da-3ee8b1a299ce.png">

I suggest you re-open the issue, because it's still not fixed in v4.9.25. You should really learn to close issues after it's confirmed that it's resolved.

**Comment by Andre601** [comment]
> <img width="1182" alt="image" src="https://user-images.githubusercontent.com/10693490/100352051-50d33500-2fe4-11eb-91da-3ee8b1a299ce.png">
> 
> @Gusted I suggest you re-open the issue, because it's still not fixed in v4.9.25. You should really learn to close issues after it's confirmed that it's resolved.

You seem to be the only one having this?
I on my end have it fixed since 4.9.24.

Did you make sure to have the CSS reset and not having any custom changes?

**Comment by kkoomen** [comment]
>  I on my end have it fixed since 4.9.24.

Can you share a picture with default settings but changing the contrast to -20? I need proof it's my mistake.

> Did you make sure to have the CSS reset and not having any custom changes?

Yes. I did do that.

**Comment by kkoomen** [comment]
I also have this problem on my other laptop. Same version, I did reset things, but still run into the same problem. I highly doubt whether this is fixed. If it really is, what am I doing wrong?

**Comment by caneleex** [comment]
> 
> 
> > <img alt="image" width="1182" src="https://user-images.githubusercontent.com/10693490/100352051-50d33500-2fe4-11eb-91da-3ee8b1a299ce.png">
> > @Gusted I suggest you re-open the issue, because it's still not fixed in v4.9.25. You should really learn to close issues after it's confirmed that it's resolved.
> 
> You seem to be the only one having this?
> I on my end have it fixed since 4.9.24.
> 
> Did you make sure to have the CSS reset and not having any custom changes?

no, he's not the only one having this issue

**Comment by Gusted** [comment]
... *Sigh* I won't even bother github.com with custom fixes I will focus on #4291 that fixes this issue globally and never look at this big pile mess again.
As why the current bug exist, IDK. Probably some selector that is doing this.

**Comment by Gusted** [comment]
> As why the current bug exist, IDK. Probably some selector that is doing this.

#4295 I was right

**Comment by kkoomen** [comment]
@Gusted I don't know how you test but it's not in being tested carefully. I use default settings but the contrast is -20. Please use this for testing purpose as it will give you a clear look on what's wrong, because I don't have issues on the issue page. It's on every page on github: homepage, repositories, issue overview, issue detail pages etc.

Please fix it and test it carefully throughout github before closing this. You should be able to run into the same problem easily so you can prevent us all from spending time on testing your fixes.

**Comment by Gusted** [comment]
The problem more relies on github.com that's now changing its CSS every day as they have announced a native dark mode. However it's still in beta and changing here and their how their CSS works. By the time I had made the fix it worked however when the release was it's was already broken as github thought it was fun to change once again their CSS logic and became indeed broken, I'm trying to tackle this problem by having a cat and mouse game with having proper variable support inside the codebase #4291. In the latest 4.9.26 their was some css change to fix the comment section and the header of an title which where indeed problematic and should be fixex(I've tested a lot of different settings).

**Comment by Gusted** [comment]
Hi everyone!

I've just finished the PR(#4291) that will solve this issue entirely without any strings attached this time.

As I've received critique before for not having the right fix nor having the edge cases covered, I would like to ask you if could side-load the PR latest binary at https://github.com/darkreader/darkreader/pull/4291#issuecomment-736090509 *Please make sure that in the dev-tools the current fix is removed.* and see if this works for you.

I've testing it so far and it didn't find weird behaviour or an inconsistency, but please report if you find one so it can be fixed.

CC @kkoomen @caneleex @privacyguy123 

Regards,
Gusted

**Comment by caneleex** [comment]
i sideloaded the build and github looks normal again for me, thanks

**Comment by kkoomen** [comment]
@Gusted I confirm that it's fixed. I did notice something else when I adjust my contrast to any value, it removes the background color at some elements. When I toggle darkreader off and on (for github) it fixes the problem. It's a new issue I guess and also small one.

Result after adjusting the contrast:

<img width="1638" alt="image" src="https://user-images.githubusercontent.com/10693490/100736375-899b5180-33ca-11eb-83f2-1c5647f200b8.png">

**Comment by Gusted** [comment]
Hi @kkoomen!

I've noticed that too, the actually is that the modified variables of Dark Reader has been for some reason removed after changing settings. I didn't find an exact reason yet, however I will have a look in the next few day. Glad to hear that it's fixed.

Regards,
Gusted

**Comment by kkoomen** [comment]
@Gusted If you can do a new release with the new build, then I'll test it again to confirm. If the new version is fixed then I'll close this issue.

**Comment by Gusted** [comment]
> @Gusted If you can do a new release with the new build, then I'll test it again to confirm. If the new version is fixed then I'll close this issue.

I will first fix the bug with changing settings and than wait for the reviews and etc. and probably another release will be released when #4291 is merged. It could be this week or next week no ETA.

**Comment by aliglelo** [comment]
I am also facing this issue.Please fix this issue

**Comment by Gusted** [comment]
> I am also facing this issue.Please fix this issue

We're working on this issue and trying to get it fixed ASAP #4291 is being reviewed to get merged.

**Comment by aliglelo** [comment]
> > I am also facing this issue.Please fix this issue
> 
> We're working on this issue and trying to get it fixed ASAP #4291 is being reviewed to get merged.

Thanks a lot.You guys are commendable.

**Comment by Andre601** [comment]
The theme is now completely broken...

![image](https://user-images.githubusercontent.com/11576465/101541764-25682700-39a2-11eb-8217-93d4fbe1cfc4.png)

But that's because GitHub now added Dark theme. Wohoo!

**Comment by Gusted** [comment]
> The theme is now completely broken...

Yep I've noticed 😞, however good news #4291 has been approved and just need some minor chances and understanding and after that for sure a new release will be made so hopefully everyone eyes doesn't need to look at this for too long.

Regards,
Gusted

**Comment by Gusted** [comment]
Hi everyone!

I just like to confirm that your old github.com & Dark Reader Color scheme experience you used to know will be possible when #4291 is merged. When you set the light theme preference in [appearance](https://github.com/settings/appearance). The dark theme preference is currently by default already having a bad contrast/non-natural color-scheme in my eyes and dark reader ruins it even further down that way and I'm not bother to have custom fixes for it.

Regards,
Gusted

**Comment by LoganDark** [comment]
RIP.

Correct:
![image](https://user-images.githubusercontent.com/4723091/101555271-29c61d00-396d-11eb-96c1-04e952eb865b.png)

Incorrect:
![image](https://user-images.githubusercontent.com/4723091/101555283-2fbbfe00-396d-11eb-9655-01b900b2a162.png)

Yuck

**Comment by kkoomen** [comment]
@Gusted @alexanderby is this published?

**Comment by Gusted** [comment]
No it isn't, but due to the mature of this PR. @alexanderby just a little version bump? I'd like to see github in normal colors again.

**Comment by kkoomen** [comment]
@Gusted @alexanderby Can you guys mention in here whether you guys released this to Firefox / Chrome for all people here? I also want to use GitHub with darkreader, because GitHub doesn't fully support darkmode for all pages >,<

**Comment by AndrewDaGuy** [comment]
> GitHub doesn't fully support darkmode for all pages >,<

@kkoomen Yep, relatable, going to try to push out a fix for some retina-burning-white images :]

**Comment by Gusted** [comment]
Heads up: Dark reader v4.9.30 has been released.
At the moment of writing it had been published on Firefox and Chrome's webstore.
